src/services/github.ts

```diff
@@ -164,6 +164,113 @@ export async function githubDelete(context: ContextType<typeof AppContext>, path
     }, {revalidate: false}); // github asks for aggressive disk caching, which we need to override.
 }
 
+// Adapted from this blog post: https://medium.com/@obodley/renaming-a-file-using-the-git-api-fed1e6f04188
+export async function githubRename(context: ContextType<typeof AppContext>, path: string, name: string) {
+    let isPublished;
+    try {
+        isPublished = context.editor.isAlreadyPublished();
+    } catch {
+        isPublished = false;
+    }
+    const pathSegments = path.split("/");
+    const basePath = dirname(path);
+    const oldName = pathSegments.at(-1);
+
+    // Ensure that another file with the same name doesn't already exist, because the renaming process will overwrite
+    // existing files otherwise
+    const current: Entry[] = context.github.cache.get(contentsPath(basePath, context.github.branch));
```

Comment:
I've now added a 30 second refresh interval to the SWR config, so the GitHub cache should stay mostly up to date. Since responses from GitHub have caching headers that ask the browser to cache them for 60s, it's not easy to get the latest data.